### PR TITLE
GUACAMOLE-623: Add support for attaching to the terminals of containers in Kubernetes pods.

### PR DESCRIPTION
**Pull request by mike-jumper**
As described in [GUACAMOLE-623](https://issues.apache.org/jira/browse/GUACAMOLE-623), this mere ~2.7K line change adds a new protocol plugin to guacamole-server, "libguac-client-kubernetes", which supports Kubernetes' web service / API.

The new plugin leverages libwebsockets and libssl to connect to the endpoint used for attaching to containers, exposing the data transferred through that endpoint through the common terminal emulator. From the user perspective, things behave essentially identically to SSH and telnet.

Sadly, the Kubernetes API documentation regarding attaching to a container does not exist. I managed to determine how the endpoint functions by:

* Looking at the "attach" implementation in Kubernetes' Java API: https://github.com/kubernetes-client/java (attach uses WebSockets for streaming)
* Looking at the WebSocket implementation within Kubernetes' Python API to determine how messages are actually framed: https://github.com/kubernetes-client/python-base (there is a concept of "channels", with the leading byte of each WebSocket message denoting the channel receiving/sending the data)
* Looking at old Kubernetes pull requests to get a rough idea of how terminal resize was implemented (it uses a separate channel and JSON message containing the new width and height)
* Verifying all the above using a tcpdump capture of traffic between kubectl and a test minikube instance with the insecure, unencrypted API server enabled

The plugin implements two methods of connecting: (1) unencrypted and without authentication, and (2) SSL/TLS with client authentication via a certificate and private key.

### CODE REVIEW

File: src/protocols/kubernetes/settings.h [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+     * recording. Key events are NOT included by default within the recording,
+     * as doing so has privacy and security implications.  Including key events
+     * may be necessary in certain auditing contexts, but should only be done
+     * with caution. Key events can easily contain sensitive information, such
+     * as passwords, credit card numbers, etc.
+     */
+    bool recording_include_keys;
+
+    /**
+     * The ASCII code, as an integer, that the telnet client will use when the
```
[the remark this answers is not in the document]
Comment (mike-jumper) [reply src/protocols/kubernetes/settings.h]: Ack! I've been found out!

[the remark this answers is not in the document]
Comment (mike-jumper) [reply src/protocols/kubernetes/settings.h]: I maaaayy have used the telnet support as a basis for this.

**Comment by jmuehlner** [comment]
I don't really understand everything that's going on here, but I don't see anything wrong.

**Comment by necouchman** [comment]
Will merge shortly.

**Comment by yinjingjiu** [comment]
Dear friend, When to release v1.1.0 ? 
 I have an kubernetes API (wss://192.168.4.102:6443/apis/subresources.kubevirt.io/v1alpha3/namespaces/default/virtualmachineinstances/testvm/vnc).
How to connect through guacamole?

**Comment by necouchman** [comment]
When all of the items in the release are ready:

https://issues.apache.org/jira/browse/GUACAMOLE/fixforversion/12343049

**Pull request merged**